fossology/fossology
# feat(reuse): Detect LICENSES folder and generate REUSE summary

Signed-off-by: Rohit Pandey <rohit.pandey4900@gmail.com>

## Description
As discussed in issue #1592 and #2214, implements a part of the REUSE.Software specification by detecting the license files added in LICENSES folder and generating REUSE summary in license browse page.

### Changes
1. Added `SearchHelperDao.php` in lib/php/dao and made changes in `src/www/ui/api/Controllers/SearchController.php`, `src/www/ui/ajax-notice-files.php` and `src/www/ui/search.php` files that uses function of this file.
2. To detect the LICENSES directory, created `DetectLicensesFolder.php` in src/lib/php/BusinessRules. The getDeclearedLicense() function returns an array with license Ids declared in LICENSES folder.
3. To compare the license findings, created `ReuseReportProcessor.php` in src/lib/php/BusinessRules. GetReuseSummary() function compares the cleared licenses and declared licenses.
4. `src/www/ui/async/AjaxReuseReport.php` handles AJAX request.

### How to test
1. Start new upload - you may want to use example package available here: https://github.com/fossology/fossology/wiki/ReSo-%28REUSE.Software%29#testing and run agents.
2. Check REUSE summary in License Browse page.

<a href="https://gitpod.io/#https://github.com/fossology/fossology/pull/2312"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
